### PR TITLE
Port image_proc test to ROS 2

### PR DESCRIPTION
**Pull request by ahcorde**

### CODE REVIEW

**Comment by ahcorde** [comment]
Relate to https://github.com/ros-perception/image_pipeline/issues/407

File: image_proc/test/rostest.cpp [hunk 1]
```diff
@@ -30,28 +30,29 @@
...
+using std::placeholders::_1;
 
 class ImageProcTest
   : public testing::Test
 {
 protected:
   virtual void SetUp()
   {
-    ros::NodeHandle local_nh("~");
+    node = rclcpp::Node::make_shared("image_proc_Test");
```
Comment (mikeferguson) [inline image_proc/test/rostest.cpp]: is there a particular reason for the capital T?

File: image_proc/test/test_rectify.cpp [hunk 2]
```diff
@@ -122,37 +129,49 @@ class ImageProcRectifyTest
...
-  void imageCallback(const sensor_msgs::ImageConstPtr & msg)
+  void imageCallback(const sensor_msgs::msg::Image::ConstSharedPtr & msg)
   {
     cv_bridge::CvImageConstPtr cv_ptr;
 
     try {
       cv_ptr = cv_bridge::toCvShare(msg, sensor_msgs::image_encodings::BGR8);
     } catch (cv_bridge::Exception & e) {
-      ROS_FATAL("cv_bridge exception: %s", e.what());
+      RCLCPP_INFO(node->get_logger(), "cv_bridge exception: '%s'", e.what());
```
Comment (mikeferguson) [inline image_proc/test/test_rectify.cpp]: this should at least be RCLCPP_ERROR (I'm not sure there is an RCLCPP_FATAL)

**Review by mikeferguson — CHANGES_REQUESTED** [review]

File: image_proc/test/test_rectify.cpp (hunk 2 above)
Comment (ahcorde) [reply image_proc/test/test_rectify.cpp]: https://github.com/ros-perception/image_pipeline/pull/910/commits/f74aa0d71e5016de8d039bf10b00ed36e14c1bd6

File: image_proc/test/rostest.cpp (hunk 1 above)
Comment (ahcorde) [reply image_proc/test/rostest.cpp]: https://github.com/ros-perception/image_pipeline/pull/910/commits/f74aa0d71e5016de8d039bf10b00ed36e14c1bd6

**Review by mikeferguson — APPROVED** [review]

**Pull request merged**